File: Frontend/lib/models/pending_player.dart
class PendingPlayer {
  final int id;
  final String nome;
  final String? soprannome;
  final String? posizione;
  final int bravura;
  final int affidabilita;
  final bool tesserato;
  final String? note;

  PendingPlayer({
    required this.id,
    required this.nome,
    this.soprannome,
    this.posizione,
    required this.bravura,
    required this.affidabilita,
    required this.tesserato,
    this.note,
  });

  factory PendingPlayer.fromJson(Map<String, dynamic> json) => PendingPlayer(
        id: json['id'] as int,
        nome: json['nome'] as String,
        soprannome: json['soprannome'] as String?,
        posizione: json['posizione'] as String?,
        bravura: (json['bravura'] as num?)?.toInt() ?? 3,
        affidabilita: (json['affidabilita'] as num?)?.toInt() ?? 3,
        tesserato: json['tesserato'] as bool? ?? false,
        note: json['note'] as String?,
      );
}

File: Backend/Models/Entities/PendingPlayer.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using CalcioAcinque.Backend.Models.Enums;

namespace CalcioAcinque.Backend.Models.Entities;

[Table("pending_players")]
public class PendingPlayer
{
    [Key]
    [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
    public int Id { get; set; }

    public int TeamId { get; set; }

    [Required, MaxLength(100)]
    public string Nome { get; set; } = string.Empty;

    [MaxLength(100)]
    public string? Soprannome { get; set; }

    public PlayerPosition? Posizione { get; set; }

    [Range(1, 5)]
    public int Bravura { get; set; } = 3;

    [Range(1, 5)]
    public int Affidabilita { get; set; } = 3;

    public bool Tesserato { get; set; } = false;

    [MaxLength(500)]
    public string? Note { get; set; }

    public bool Claimed { get; set; } = false;
    public int? ClaimedByPlayerId { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    [ForeignKey("TeamId")]
    public virtual Team Team { get; set; } = null!;
}
